File: hw/ramos/u3/hw28PrinciplesAplication/mongoPyhton/controller/MongoCrud.py
from controller.MongoConnection import MongoConnection
from model.Store import Store
from pymongo import MongoClient

class MongoCrud:

    def __init__(self):
        db = MongoConnection.get_database()
        self.collection = db["store"]

    # CREATE
    def create(self, store: Store):
        store.price_iva = store.calculate_price_iva()
        self.collection.insert_one({
            "id": store.id,
            "name": store.name,
            "price": store.price,
            "priceIva": store.price_iva
        })

    # READ 
    def read_by_id(self, id):
        d = self.collection.find_one({"id": id})
        if d is None:
            return None

        return Store(
            d["id"],
            d["name"],
            d["price"],
            d["priceIva"]
        )

    # UPDATE
    def update(self, store: Store):
        store.price_iva = store.calculate_price_iva()
        self.collection.update_one(
            {"id": store.id},
            {"$set": {
                "name": store.name,
                "price": store.price,
                "priceIva": store.price_iva
            }}
        )

    # DELETE
    def delete(self, id):
        self.collection.delete_one({"id": id})

    def read_all(self):
        stores = []

        cursor = self.collection.find()  # 👈 find(), NO find_one()

        for d in cursor:
            store = Store(
                d["id"],
                d["name"],
                d["price"],
                d["priceIva"]
            )
            stores.append(store)

        return stores

File: hw/ramos/u3/hw28PrinciplesAplication/mongoJavaScript/controller/MongoCrud.js
const Store = require("../model/Store");

class MongoCrud {

  async create(store) {
    store.priceIva = store.price * 1.12;
    return await Store.create(store);
  }

  async readAll() {
    const stores = await Store.find();
    return stores.map(s => ({
      id: s.id,
      name: s.name,
      price: s.price,
      priceIva: s.price * 1.12
    }));
  }

  async readById(id) {
    const s = await Store.findOne({ id });
    if (!s) return null;

    return {
      id: s.id,
      name: s.name,
      price: s.price,
      priceIva: s.price * 1.12
    };
  }

  async update(store) {
    await Store.updateOne(
      { id: store.id },
      { $set: { name: store.name, price: store.price } }
    );
    return true;
  }

  async delete(id) {
    await Store.deleteOne({ id });
    return true;
  }
}

module.exports = MongoCrud;

